diff --git a/app/src/lib/helpers/linux.ts b/app/src/lib/helpers/linux.ts
index e6e021808..87f8214f7 100644
--- a/app/src/lib/helpers/linux.ts
+++ b/app/src/lib/helpers/linux.ts
@@ -22,15 +22,21 @@ export function convertToFlatpakPath(path: string) {
     return path
   }
 
-  if (path.startsWith('/opt/')) {
+  if (path.startsWith('/opt/')|| path.startsWith("/var/lib/flatpak")) {
     return path
   }
-
   return join('/var/run/host', path)
 }
 export function formatWorkingDirectoryForFlatpak(path: string): string {
   return path.replace(/(\s)/, "\ ")
 }
+export function formatPathForFlatpak(path:string):string{
+  if (path.startsWith("/var/lib/flatpak/app")){
+    return path.replace("/var/lib/flatpak/app/","")
+  }else{
+    return path
+  }
+}
 /**
  * Checks the file path on disk exists before attempting to launch a specific shell
  *
@@ -84,10 +90,11 @@ export function spawnEditor(
   options: SpawnOptions
 ): ChildProcess {
   if (isFlatpakBuild()) {
+    let actualpath = formatPathForFlatpak(path)
     let EscapedworkingDirectory = formatWorkingDirectoryForFlatpak(workingDirectory)
     return spawn(
       'flatpak-spawn',
-      ['--host', path, EscapedworkingDirectory],
+      ['--host', actualpath, EscapedworkingDirectory],
       options
     )
   } else {
